spec/compgen.test.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+#### Print function list
+env -i /bin/bash --norc --noprofile -c 'add () { expr 4 + 4; }; \
```

Comment:
OSH isn't being tested here, because you're invoking bash.

You can probably do something with the value of `$SH`.  Or maybe we can add something to `sh_spec.py` that starts bash with a clean environment?

That is probably good for test isolation in general.